addon/routes/mymaterials.js

```diff
@@ -0,0 +1,28 @@
+import Route from '@ember/routing/route';
+import { reads } from '@ember/object/computed';
+import { inject as service } from '@ember/service';
+import AuthenticatedRouteMixin from 'ember-simple-auth/mixins/authenticated-route-mixin';
+import { task } from 'ember-concurrency';
+
+export default Route.extend(AuthenticatedRouteMixin, {
+  commonAjax: service(),
+  currentUser: service(),
+  iliosConfig: service(),
+
+  host: reads('iliosConfig.apiHost'),
+  namespace: reads('iliosConfig.apiNameSpace'),
+
+  model() {
+    return this.fetchModelData.perform();
+  },
+
+  fetchModelData: task(function* () {
```

Comment:
That's how many people pull route data via ember-concurrency?

I could restore the below code if you want.
```
   const commonAjax = this.commonAjax;
    const host = this.host;
    const namespace = this.namespace;
    return new Promise(resolve => {
      this.currentUser.get('model').then(user => {
        const userId = user.get('id');
        let url = `${host}/${namespace}/usermaterials/${userId}`;
        commonAjax.request(url).then(data => {
          resolve(data.userMaterials);
        });
      });
    });
```

Reply:
I guess I just don't see the need here since we're not calling a `set` anywhere. For clarity you could rewrite with `async/await` on the model function.